src/box/tuple.c

```diff
@@ -179,17 +232,31 @@ tuple_extract_key_raw(const char *data, const char *data_end,
 	uint32_t current_fieldno = 0;
 	for (uint32_t i = 0; i < key_def->part_count; i++) {
 		uint32_t fieldno = key_def->parts[i].fieldno;
+		for (; i < key_def->part_count - 1; i++){
+			if (key_def->parts[i].fieldno + 1 !=
+				key_def->parts[i + 1].fieldno) {
+			break;
```

Comment:
Same (about `break` and comment). Maybe it can be moved in a special function? - it is already third copypaste of the same code.